### DISCUSSION TITLE
Fields should be added to posted tables

### DISCUSSION THREAD

**Discussion by CMouat** [StefanMaron/BusinessCentral.LinterCop#790] [New Rule]
If a field is added to a table such as sales header, the field should also be added to the posted tables to avoid future issues with fields of different types being added and erroring during transfer fields.

**Comment by StefanMaron** [comment]
@Arthurvdv  I feel like rules that are more like implementation recommendations like this one and not strictly related to the Language itself would be a good fit for a second analyzer, what do you think?

**Reply by Arthurvdv** [reply]
If we’re going down the road for multiple analyzers, it would make sense for me to do this thorough.

I could image something like this:

> StyleCop
> LinterCop
> SecurityCop
> PerformanceCop
> DocumentationCop
> ImplementationCop

There are some challenges, for example we have now one monolithic single .dll file, where in a scenario of multiple analyzers it would be recommended to have a common library to share certain methods between these analyzers. An example is the handling on objects that are marked as obsolete in AL.

It could the be opportune to review the distribution of assets/artifacts .dll file and maybe move to NuGet packages or something?

So I think if we commit to this, we need immediately make a big leap so we have a solid structure to make multiple code analyzers possible.
